### PR TITLE
Adds specific icon for each card

### PR DESCRIPTION
**Pull request by MariusCasvean**

### CODE REVIEW

File: src/views/client-dashboard/ClientPayments.vue [hunk 1]
```diff
@@ -10,7 +10,34 @@
...
+              <img
+                v-if="['mastercard', 'MasterCard'].includes(card.brand)"
+                src="@/assets/svg/mastercard.svg"
+                alt="mastercard"
+                width="30"
+                height="30"
+                class="lazy ml-n1 mr-1"
+              >
+              <img
+                v-else-if="['visa', 'Visa'].includes(card.brand)"
```
Comment (norbertdemeny13) [inline src/views/client-dashboard/ClientPayments.vue:23]: same here as for mastercard

File: src/views/client-dashboard/ClientPayments.vue [hunk 2]
```diff
@@ -10,7 +10,34 @@
...
+                src="@/assets/svg/visa.svg"
+                alt="visa"
+                width="30"
+                height="30"
+                class="lazy ml-n1 mr-1"
+              >
+              <img
+                v-else-if="['amex', 'American Express'].includes(card.brand)"
```
Comment (norbertdemeny13) [inline src/views/client-dashboard/ClientPayments.vue:31]: same here as for mastercard

File: src/views/client-dashboard/ClientPayments.vue [hunk 3]
```diff
@@ -10,7 +10,34 @@
...
+              >
+              <img
+                v-else-if="['visa', 'Visa'].includes(card.brand)"
+                src="@/assets/svg/visa.svg"
+                alt="visa"
+                width="30"
+                height="30"
+                class="lazy ml-n1 mr-1"
+              >
+              <img
```
Comment (norbertdemeny13) [inline src/views/client-dashboard/ClientPayments.vue:30]: From what I see, there are 3 type of options, but the structure is identical. So what you could do, is to have a computed property or a method, that is creating the 3 variants for you: 

such as:

`getCardInfo.src`, `getCardInfo.alt`, where in the getCardInfo, you can do:

`getCardInfo() {
if or switch - add the logic that you have in the template
Based on the logic, return the correct src and alt
`


}

File: src/views/client-dashboard/ClientPayments.vue [hunk 4]
```diff
@@ -10,7 +10,34 @@
             :class="`${card.primary ? 'active' : ''} client-payment-card-item`"
             @click.prevent="onSelect(card)"
           >
-            <div class="card-number" v-html="getCardInfo(card)" />
+            <div class="d-flex align-items-center">
+              <img
+                v-if="['mastercard', 'MasterCard'].includes(card.brand)"
```
[the remark this answers is not in the document]
Comment (MariusCasvean) [reply src/views/client-dashboard/ClientPayments.vue]: good point. Did it

File: src/views/client-dashboard/ClientPayments.vue (hunk 2 above)
Comment (MariusCasvean) [reply src/views/client-dashboard/ClientPayments.vue:31]: here seems to be another scenario. The options are spelled different.

File: src/views/client-dashboard/ClientPayments.vue (hunk 3 above)
Comment (MariusCasvean) [reply src/views/client-dashboard/ClientPayments.vue]: ok, I did some changes here also. Now we have just a single  tag with dynamic :src and :alt

File: src/views/client-dashboard/ClientPayments.vue [hunk 5]
```diff
@@ -137,7 +172,27 @@
...
       getCardInfo(card) {
-        return `&#128179; ${card.brand && card.brand.toUpperCase()} **** ${card.last4} (exp: ${card.exp_month}/${card.exp_year})`;
+        const brand = ['amex', 'American Express'].includes(card.brand) ? 'AMEX' : card.brand.toUpperCase();
+
+        return `${brand} **** ${card.last4} (exp: ${card.exp_month}/${card.exp_year})`;
+      },
+
+      getImagePath(brand) {
+        const path = require.context('@/assets/svg');
+        let svg;
```
Comment (norbertdemeny13) [inline src/views/client-dashboard/ClientPayments.vue:182]: How about:

`const localBrand = brand.toLowerCase();`

This way you will remove the 'Visa' and 'MasterCard' option.

Also, whenever you have a case, where 4 times you want to do something, 1 time something else, probably would make sense to just:

`const localBrand = brand.toLowerCase();`

`const path = ['visa', 'mastercard'].includes(localBrand) ? `./${localBrand}.svg` : './amex.svg'

File: src/views/client-dashboard/ClientPayments.vue (hunk 2 above)
Comment (norbertdemeny13) [reply src/views/client-dashboard/ClientPayments.vue]: true

File: src/views/client-dashboard/ClientPayments.vue (hunk 3 above)
Comment (norbertdemeny13) [reply src/views/client-dashboard/ClientPayments.vue]: great

File: src/views/client-dashboard/ClientPayments.vue (hunk 5 above)
Comment (norbertdemeny13) [reply src/views/client-dashboard/ClientPayments.vue]: great

**Comment by norbertdemeny13** [comment]
So far so good :)

**Comment by norbertdemeny13** [comment]
Looks good. But your implementation is restricted to the client payments view only. If you check the codebase, you'll see the card component is reused in multiple places. Make sure the icons are visible everywhere. 

Subscription Payment
Reservation Payment
Client Profile

**Comment by MariusCasvean** [comment]
>

**Comment by MariusCasvean** [comment]
as I spoke with Gabriela, for the payment pages, where we have a <select> tag with <option>, we just remove the yellow icon and leave text only. The icons will be visible, for now, only in Contul meu -> Carduri

**Comment by elroscrypto** [comment]
Looks good. you can merge it

**Pull request merged**